src/Avalonia.Controls/Primitives/PopupPositioning/IPopupPositioner.cs

```diff
@@ -443,19 +491,35 @@ public interface IPopupPositioner
         void Update(PopupPositionerParameters parameters);
     }
 
-    [Unstable]
-    static class PopupPositionerExtensions
+    internal static class PopupPositionerExtensions
```

Comment:
New file too. Convention across all of .net these days is 1-file per type.

Reply:
I don't feel about moving pre-existing types to their own files in this PR. 

But I will move CustomPopupPlacement and CustomPopupPlacementCallback to their own file, thanks. And PopupPositionRequest.

Reply:
Done

Reply:
> I don't feel about moving pre-existing types to their own files in this PR.

That's fair. Someday I want to go into a hardcore refactor of the entire controls directory. Model it based on WinUI. I haven't been successful in that discussion so far though... :)